libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs

```diff
@@ -318,5 +320,20 @@ private static T SafeCast<T>(object value)
 
             return obj.ToObject<T>();
         }
+
+        private class InvokeResponseException : Exception
+        {
+            HttpStatusCode StatusCode { get; set; }
+
+            public InvokeResponseException(HttpStatusCode statusCode)
+            {
+                StatusCode = statusCode;
+            }
+            
+            public InvokeResponse CreateInvokeResponse()
+            {
+                return new InvokeResponse { Status = (int)StatusCode };
```

Comment:
so we could add message text in here... not blocking this PR as this is all internal